frontend/src/app/components/comparison/comparison.component.ts

```diff
@@ -0,0 +1,183 @@
+      "brief": "",
+      "remarks": "",
+      "address": {
+          "id": 0,
+          "street": "",
+          "city": "",
+          "country": "",
+          "zipCode": 0
+      }
+    }
+
+    this.chartOptions = {
+      title: {
+        text: "Stock Value Comparison"
+      },
+      xAxis: {
+        labels: {
+          formatter: function () {
+            var label = this.axis.defaultLabelFormatter.call(this).replace(" ","").replace("k","000").replace(" ","").replace(" ","").replace("M", "000000");
+            var date: Date = new Date(parseInt(label));
+            return date.getDate()+"/"+date.getMonth()+"/"+date.getFullYear()+"\n"+date.getHours()+":"+date.getMinutes()+":"+date.getSeconds();
+          }
+        }
+      },
+      yAxis: {
+        title: {
+          text: "Stock Price"
+        }
+      },
+      series: [],
+      legend: {
+        title: {
+            text: 'Stock<br/><span style="font-size: 9px; color: #666; font-weight: normal">(Click to hide)</span>',
+            style: {
+                fontStyle: 'italic'
+            }
+        },
+        layout: 'vertical',
+        align: 'right',
+        verticalAlign: 'top',
+        x: -10,
+        y: 100
+      }
+    }
+
+  }
+
+  ngOnInit(): void {
+    this.state = this.authService.getCurrentUserRole();
+    this.companyService.getAllCompanies().subscribe(companies => {
+      this.companies = companies;
+    });
+    this.exchangeService.getAllExchanges().subscribe(exchanges => {
+      this.exchanges = exchanges;
+    });
+  }
+
+  onCompanyClick(company:Company){
+    this.companyTitle = company.name;
+    this.companySelected = company;
+  }
+
+  onExchangeClick(exchange:Exchange){
+    this.exchangeTitle = exchange.name;
+    this.exchangeSelected = exchange;
+  }
+
+  onSubmit(){
+    this.fromTime+= ".000+05:30";
+    this.toTime+= ".000+05:30";
+    this.stockPriceService.getStockPrices(this.companySelected.id, this.exchangeSelected.id, this.fromTime, this.toTime).subscribe( stockPrices => {
+      if(stockPrices.length){
+        this.stockPrices[this.stockPrices.length] = stockPrices;
+        console.log(this.stockPrices);
+        this.chartOptions.series?.push(this.getPriceSeries(stockPrices));
+        Highcharts.chart('chart-container', this.chartOptions);
+      } else{
+        alert("No data found for the requested period");
+      }
+    })
+    this.onReset();
+  }
+
+  onReset(){
+    this.companySelected = {
+      "id": 0,
+      "name": "",
+      "turnover": 0,
+      "ceo": "",
+      "brief": "",
+      "bod": "",
+      "sector": {
+          "id": 0,
+          "name": "",
+          "brief": ""
+      }
+    }
+    this.exchangeSelected = {
+      "id": 0,
+      "name": "",
+      "brief": "",
+      "remarks": "",
+      "address": {
+          "id": 0,
+          "street": "",
+          "city": "",
+          "country": "",
+          "zipCode": 0
+      }
+    }
+    this.companyTitle='Please choose a company';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ibmpariwesh/socgen-stockmarket-project/pullRequest?prid=8405405)
```suggestion
    this.companyTitle="Please choose a company";
```